src/index.js

```diff
@@ -51,14 +55,15 @@ const updateWithFrame = (
 ): Promise<void> => {
   if (frame.kind === 'frame.yield') {
     return new Promise((resolve, reject) => {
-      setImmediate(() => {
+      const resume = () => {
         try {
           resumeWithDispatcher(frame, queue, visitor)
           resolve()
         } catch (err) {
           reject(err)
         }
-      })
+      }
+      isServer ? setImmediate(resume) : resume()
```

Comment:
I know this is a nit (apart from the variable name change) but the minifier should take care of turning if-statements into ternary-expressions, but otherwise I think this codebase prefers the former 😄

```suggestion
      if (SHOULD_YIELD) {
        setImmediate(resume)
      } else {
        resume()
      }
```